internal/agent/device/bootstrap.go

```diff
@@ -110,6 +120,68 @@ func (b *Bootstrap) Initialize(ctx context.Context) error {
 	return nil
 }
 
+func (b *Bootstrap) EnsureCurrentRenderedSpecUpToDate(ctx context.Context) error {
```

Comment:
just curious why you want to export these the bootstrap object will only ever call Initialize I would like to keep these local until we have an explicit usecase IE integration tests requires etc. I just don't want to give the impression that bootstrap is portable.

Reply:
I wanted the unit tests to be in a separate package, but you're right.  I moved the unit tests back to the same package.